compiler/rustc_ty_utils/messages.ftl

```diff
@@ -12,7 +12,7 @@ ty_utils_array_not_supported = array construction is not supported in generic co
 
 ty_utils_block_not_supported = blocks are not supported in generic constants
 
-ty_utils_never_to_any_not_supported = converting nevers to any is not supported in generic constants
+ty_utils_never_to_any_not_supported = converting never to any is not supported in generic constants
```

Comment:
I don't exactly love this change but on the other hand, this error message is very bad anyways and I don't think this makes it worse, so the change can be kept. I'm not sure whether any of this can be hit on stable anyways, I think this should all be behind `generic_const_exprs` so it's fine.

Reply:
Would ``coercing the `never` type is not supported in generic constants`` make more sense?

Reply:
That sounds good, could you add that in a separate commit?